### PR TITLE
Accept Promise as Validator.message|missing_message

### PR DESCRIPTION
**Pull request by q0w**
# I have made things!
https://github.com/encode/django-rest-framework/blob/master/rest_framework/validators.py

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]

File: rest_framework-stubs/validators.pyi [hunk 1]
```diff
@@ -2,6 +2,7 @@ from collections.abc import Callable, Container, Iterable, MutableMapping
 from typing import Any, Protocol, TypeVar
 
 from django.db.models import Model, QuerySet
+from django.utils.functional import _StrOrPromise
```
Comment (intgr) [inline rest_framework-stubs/validators.pyi:5]: Use `from django_stubs_ext import StrOrPromise`, which is public API.

File: pyproject.toml [hunk 2]
```diff
@@ -14,6 +14,7 @@ extend-exclude = [
 line-length = 120
 target-version = "py38"
 # See Rules in Ruff documentation: https://docs.astral.sh/ruff/rules/
+[tool.ruff.lint]
```
Comment (intgr) [inline pyproject.toml:17]: Would you mind opening a separate MR about this?

**Review by intgr — CHANGES_REQUESTED** [review]

File: rest_framework-stubs/validators.pyi (hunk 1 above)
Comment (q0w) [reply rest_framework-stubs/validators.pyi:5]: I checked django-stubs for that

File: pyproject.toml (hunk 2 above)
Comment (q0w) [reply pyproject.toml:17]: Do you want to revert these changes?

Comment (intgr) [reply pyproject.toml:17]: No, I was seconds late with my review. No reason to revert if it's merged already.

File: rest_framework-stubs/validators.pyi (hunk 1 above)
Comment (intgr) [reply rest_framework-stubs/validators.pyi:5]: Yeah, that's internal to `django-stubs`. Other projects should import from `django_stubs_ext`

Comment (q0w) [reply rest_framework-stubs/validators.pyi:5]: https://github.com/typeddjango/djangorestframework-stubs/pull/621

**Pull request merged**